JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/GroupBean.java

```diff
@@ -146,31 +136,160 @@ private int getUserId(String userNickName) {
+        }catch(NoResultException noResult) {
+            logger.log(Level.WARNING, "NO RESULT IN QUERY IN GETACCESS()", noResult);
+            return null;
+        }
         return access;
     }
 
-    private Group getGroup(int groupId) {
+    public Group getGroup(int groupId) {
+
+        Group group = null;
+        try {
+            group = (Group) em.createQuery("SELECT g FROM Group g WHERE g.groupId=:groupid")
+                    .setParameter("groupid", groupId)
+                    .getSingleResult();
+        }catch(NoResultException noResult) {
+            logger.log(Level.WARNING, "NO RESULT IN QUERY IN GETGROUP()", noResult);
+            return null;
+        }
+
+        return group;
+    }
+
+    public Group getGroup(String groupName) {
+
+        Group group = null;
+        try {
+            group = (Group) em.createQuery("SELECT g FROM Group g WHERE g.groupName=:groupname")
+                    .setParameter("groupname", groupName)
+                    .getSingleResult();
+        }catch(NoResultException noResult) {
+            logger.log(Level.WARNING, "NO RESULT IN QUERY IN GETGROUP()", noResult);
+            return null;
+        }
+        return group;
+    }
+
+    public int getGroupId(String groupName) {
+
+        Group group = getGroup(groupName);
+        if (group == null) {
+            return -1;
+        }
+        return group.getGroupId();
+
+    }
+
+    public Group createGroup(String groupName) {
+
+        Group g = new Group(groupName);
+
+        em.persist(g);
+        em.flush();
+
+        return g;
+
+
+    }
+
+    public Access createAccess(String accessName) {
+        Access a = new Access(accessName);
+
+        em.persist(a);
+        em.flush();
+
+        return a;
+
+    }
+
+    public UserGroup createUserGroup(User u, Group g, Access a) {
+
+
+/*
+        Access a = (Access)em.createQuery("SELECT a FROM Access a WHERE a.accessId =:accessid")
+                            .setParameter("accessid", accessId)
+                            .getSingleResult();
 
-        Group group = (Group)em.createQuery("SELECT g FROM Group g WHERE g.groupId=:groupid")
+        Group g = (Group)em.createQuery("SELECT g FROM Group g WHERE g.groupId =:groupid")
                 .setParameter("groupid", groupId)
                 .getSingleResult();
-        return group;
+
+        User u = (User)em.createQuery("SELECT u FROM User u WHERE u.userId =:userid")
+                .setParameter("userid", userId)
+                .getSingleResult();*/
+
+        UserGroup ug = new UserGroup(g, u, a);
+
+        ug.setUserId(u.getUserId());
+        ug.setGroupId(g.getGroupId());
+
+        //  em.getTransaction().begin();
+        em.persist(ug);
+        //       em.flush();
+        //  em.getTransaction().commit();
+        return ug;
+
     }
 
-    //!TODO
-    public Map<String, String> getAllMembers(int groupId){
-        return null;
+    public UserGroup getUserGroup(int userId, int groupId) {
```

Comment:
Isn't it bad to have much interference between beans? I thought that one small supportive methods could repeat.

23 марта 2016 г., в 2:25, alexander1leonov <notifications@github.com> написал(а):

> In JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/GroupBean.java:
> 
> >      }
> >  
> > -    //!TODO
> > -    public Map<String, String> getAllMembers(int groupId){
> > -        return null;
> > +    public UserGroup getUserGroup(int userId, int groupId) {
> TODO: move UserGroup operations to UserGroupBean
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>